network/iptables_restore.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2015 flannel authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// +build !windows
+
+package network
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/coreos/go-iptables/iptables"
+	log "k8s.io/klog"
+)
+
+const (
+	ipTablesRestoreCmd  string = "iptables-restore"
+	ip6TablesRestoreCmd string = "ip6tables-restore"
+)
+
+// IPTablesRestore wrapper for iptables-restore
+type IPTablesRestore interface {
+	// ApplyFully apply all rules and flush chains
+	ApplyFully(rules IPTablesRestoreRules) error
+	// ApplyPartial apply without flush chains
+	ApplyPartial(rules IPTablesRestoreRules) error
```

Comment:
Maybe call this `ApplyNoFlush` or `ApplyWithoutFlush`?

Reply:
:ok: